save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/utils/Warning.kt

```diff
@@ -36,35 +39,73 @@ data class Warning(
 @Suppress(
     "TooGenericExceptionCaught",
     "SwallowedException",
-    "ThrowsCount")
+    "ThrowsCount",
+    "TOO_LONG_FUNCTION",
+    "ReturnCount",
+    "ComplexMethod",
+    "NestedBlockDepth",
+    "AVOID_NULL_CHECKS",
+)
 internal fun String.extractWarning(warningRegex: Regex,
                                    fileName: String,
                                    line: Int?,
                                    columnGroupIdx: Int?,
                                    messageGroupIdx: Int,
 ): Warning? {
-    val groups = warningRegex.find(this)?.groups ?: return null
-
-    val column = columnGroupIdx?.let {
-        try {
-            groups[columnGroupIdx]!!.value.toInt()
-        } catch (e: Exception) {
-            throw ResourceFormatException("Could not extract column number from line [$this], cause: ${e.message}")
-        }
+    val defaultLinePattern = Regex("// ;warn: (.*)")
+    val groups = warningRegex.find(this)?.groups
+    val defaultLineGroups = if (groups == null) {
+        defaultLinePattern.find(this)?.groups
+    } else {
+        null
     }
 
-    val message = try {
-        groups[messageGroupIdx]!!.value
-    } catch (e: Exception) {
-        throw ResourceFormatException("Could not extract warning message from line [$this], cause: ${e.message}")
+    if (groups == null && defaultLineGroups == null) {
+        return null
     }
 
-    return Warning(
-        message,
-        line,
-        column,
-        fileName,
-    )
+    groups?.let {
+        val column = columnGroupIdx?.let {
+            try {
+                val colValue = groups[columnGroupIdx]!!.value
+                if (colValue.isEmpty()) {
+                    null
+                } else {
+                    colValue.toInt()
+                }
+            } catch (e: Exception) {
+                throw ResourceFormatException("Could not extract column number from line [$this], cause: ${e.message}")
+            }
```

Comment:
Let's create a utility method `getRegexGroupSafe(idx: Int?)` which will encapsulate this `try-catch` and null handling